control/controlexecute/control_run.go

```diff
@@ -147,24 +169,31 @@ func (r *ControlRun) Execute(ctx context.Context, client db_common.Client) {
 			r.group.updateSeverityCounts(r.Severity, r.Summary)
 		}
 		r.Lifecycle.Add("execute_end")
-		r.Duration = r.Lifecycle.GetDuration()
+		r.Duration = time.Since(startTime)
+		if r.group != nil {
+			r.group.addDuration(r.Duration)
+		}
 		log.Printf("[TRACE] finishing with concurrency, %s, , %d\n", r.Control.Name(), r.executionTree.progress.executing)
 	}()
 
+	if utils.IsContextCancelled(ctx) {
```

Comment:
why do we need this here? how did it work before?

Reply:
earlier, the ones which were waiting for connection  would report a `error acquiring database connection` error.

with this change, it reports the proper `execution cancelled` error

Reply:
my mistake - there's no point in putting it there